### ISSUE TITLE
Design: table vs treeview

### ISSUE THREAD

**Issue by fcollonval** [lckr/jupyterlab-variableInspector#33]
I'd like to start a discussion about the layout to use.
I fully understand why Matlab-like softwares use table like layout as they focus on values and originally do not have object-oriented data structure. But this is not the case in general for Python and lots of other languages.
So a tree view, like the one used in Pycharm could be a more general solution. 
![variables pane](
https://www.jetbrains.com/help/img/idea/2018.1/py_view_as_array.png)

In particular, basic types are directly seen in the tree. And array can be seen in details in side panels as you have done.
The object attributes will be looked for only when the user is expanding the object leaf.

**Comment by chappers** [comment]
Yes, this is very similar to the RStudio variable explorer as well, where objects can be expanded. I'll leave for lckr to give their view. I think its something we can implement into the future, with an eye on how we _might_ do it (basically its picking up the `dir` output of everything variable). 

My personal view is that although valuable, it is probably less valuable than:  
*  Sorting out support and generalising for other languages (what about Matlab/Octave kernel? this context wouldn't be useful there as you've pointed out)
*  Making it easier for the community to extend this variable inspector

Of course if you know how to extend it yourself, I would absolutely encourage you to submit a PR.

**Comment by fcollonval** [comment]
Thanks for the feedback.

On the front for multi languages support, there is some work on the jupyter
core to test VS Code protocol for debugging - and so inspect variables. It
maybe worth trying connecting the dots.

Here is the link https://github.com/jupyter/debugger/blob/master/README.md

>

**Comment by chappers** [comment]
Given who we are in this project, I doubt we have the clout to influence future architectural decisions related to Jupyter...

Since that project is talking about large architectural decisions (_potentially_ breaking this extension + many other extensions) and roadmaps years down the track - I'll be leaving it alone for now.
